packages/@aws-cdk/aws-s3/lib/bucket.ts

```diff
@@ -1874,11 +1874,10 @@ export class Bucket extends BucketBase {
     }
 
     rule.transitions?.forEach((transition) => {
-      if (transition.transitionAfter === undefined && transition.transitionDate === undefined) {
-        throw new Error('Exactly one of transitionDate and transitionAfter must be specified');
-      }
-
-      if (transition.transitionAfter !== undefined && transition.transitionDate !== undefined) {
+      if (
+        (transition.transitionAfter === undefined && transition.transitionDate === undefined) ||
+        (transition.transitionAfter !== undefined && transition.transitionDate !== undefined)
```

Comment:
The condition can be simplified and made to check for empty entries too:
```
((transition.transitionAfter && transition.transitionDate) || (!transition.transitionAfter && !transition.transitionAfter))
```
***But***, being verbose makes it easier to troubleshoot; if the message states clearly you either have none or both specified.